DESIGN_PRINCIPLES.md

```diff
@@ -0,0 +1,32 @@
+# Design principles
+
+This documents aims to capture the design principles that went into this library.
+It should serve as a reference point when making decisions on what features to include or exclude.
+
+## Simple
+
+One of most important goals that we want to adhere to is creating a _simple_ API.
+Overall, this means keeping the API as small as possible to get the task done.
+When in doubt, we'd rather not add flags or configuration options for certain use cases.
+
+Tests should be easier to write, easy to understand and easy to maintain.
+`testcontainers` aims to support this as much as possible.
```

Comment:
The umbrella project is called `Testcontainers`. The implementation for the individual languages is called `Testcontainers for {$lang}` and the repository is called `testcontainers-${rust}`. 

In think in this instance, it is indeed referring to Testcontainers in general and to Testcontainers for Rust in particular 🙂

Reply:
I intended to refer to the crate name here :D

Would you like it to be rewritten?